pkg/resources/masking_policy.go

```diff
@@ -2,282 +2,339 @@ package resources
+	client := meta.(*provider.Context).Client
+	id, err := sdk.ParseSchemaObjectIdentifier(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	policy, err := client.MaskingPolicies.ShowByID(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if err := d.Set("name", id.Name()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("database", id.DatabaseName()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("schema", id.SchemaName()); err != nil {
+		return nil, err
+	}
+	if err := d.Set("exempt_other_policies", booleanStringFromBool(policy.ExemptOtherPolicies)); err != nil {
+		return nil, err
+	}
+	policyDescription, err := client.MaskingPolicies.Describe(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	if err := d.Set("body", policyDescription.Body); err != nil {
+		return nil, err
+	}
+	if err := d.Set("argument", schemas.MaskingPolicyArgumentsToSchema(policyDescription.Signature)); err != nil {
+		return nil, err
+	}
+	return []*schema.ResourceData{d}, nil
+}
+
+func CreateMaskingPolicy(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
 
 	name := d.Get("name").(string)
 	databaseName := d.Get("database").(string)
 	schemaName := d.Get("schema").(string)
+	id := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
 
-	expression := d.Get("masking_expression").(string)
+	expression := d.Get("body").(string)
 	returnDataType := d.Get("return_data_type").(string)
 
-	ctx := context.Background()
-	objectIdentifier := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
-
-	signatureList := d.Get("signature").([]interface{})
-	signature := []sdk.TableColumnSignature{}
-	for _, s := range signatureList {
-		m := s.(map[string]interface{})
-		columns := m["column"].([]interface{})
-		for _, c := range columns {
-			cm := c.(map[string]interface{})
-			dt, err := sdk.ToDataType(cm["type"].(string))
-			if err != nil {
-				return err
-			}
-			signature = append(signature, sdk.TableColumnSignature{
-				Name: cm["name"].(string),
-				Type: dt,
-			})
+	arguments := d.Get("argument").([]any)
+	args := make([]sdk.TableColumnSignature, 0)
+	for _, arg := range arguments {
+		v := arg.(map[string]any)
+		dataType, err := sdk.ToDataType(v["type"].(string))
+		if err != nil {
+			return diag.FromErr(err)
 		}
+		args = append(args, sdk.TableColumnSignature{
+			Name: v["name"].(string),
+			Type: dataType,
+		})
 	}
 
 	returns, err := sdk.ToDataType(returnDataType)
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
+
+	// set optionals
 	opts := &sdk.CreateMaskingPolicyOptions{}
 	if comment, ok := d.Get("comment").(string); ok {
 		opts.Comment = sdk.String(comment)
 	}
-	if exemptOtherPolicies := d.Get("exempt_other_policies").(bool); exemptOtherPolicies {
-		opts.ExemptOtherPolicies = sdk.Bool(exemptOtherPolicies)
+	if v := d.Get("exempt_other_policies").(string); v != BooleanDefault {
+		parsed, err := strconv.ParseBool(v)
```

Comment:
we have custom function for bool value parsing

Reply:
Changed.